sia/src/rhp/merkle.rs

```diff
@@ -73,6 +75,149 @@ pub fn sector_root(sector: &[u8]) -> Hash256 {
     )
 }
 
+#[derive(Debug, Error)]
+pub enum ProofValidationError {
+    #[error("Invalid proof length: expected {expected}, got {actual}")]
+    InvalidProofLength { expected: usize, actual: usize },
+
+    #[error("Invalid proof root: expected {expected}, got {actual}")]
+    InvalidProofRoot { expected: Hash256, actual: Hash256 },
+
+    #[error("IO error: {0}")]
+    Io(#[from] io::Error),
+
+    #[error("The proof data is not segment aligned")]
+    NotSegmentAligned,
+}
+
+pub type Result<T> = std::result::Result<T, ProofValidationError>;
+
+pub struct RangeProof;
```

Comment:
You could consider making the proof hashes an unamed field here instead of a param on `verify` (`pub struct RangeProof(Vec<Hash256>)`) and derive AsyncSiaEncodable/AsyncSiaDecodable. You could then change the return type of RPCReadResponse.proof to `RangeProof`.